### PR TITLE
Added support for SubscriptionPurchaseOptions in ProductResponses

### PR DESCRIPTION
**Pull request by nicfix**
## Motivation / Description
We updated the /products endpoint to return purchase options that include, as example, trial offers.
This PR adds the necessary interfaces to support the data returned by Khepri as of now.

## Changes introduced
* Added the new interfaces in product-responses.ts
* Updated the test response from the endpoint with a trial offer and a base option

## Linear ticket
BIL-502

### CODE REVIEW

File: src/networking/responses/products-response.ts [hunk 1]
```diff
@@ -1,3 +1,23 @@
...
+  id: string;
+}
+
+export interface SubscriptionPurchaseOption extends PurchaseOption {
+  base_price: PurchaseOptionPrice;
+  trial: PurchaseOptionPrice | null;
+}
+
 export interface ProductResponse {
   current_price: {
```
Comment (tonidero) [inline src/networking/responses/products-response.ts:22]: You could potentially substitute the type here to `Price` I think? Though we will be mostly ignoring `current_price` in future PRs I imagine and we could even remove it then.

File: src/networking/responses/products-response.ts [hunk 2]
```diff
@@ -1,3 +1,23 @@
+export interface Price {
+  amount: number;
+  currency: string;
+}
+
+export interface PurchaseOptionPrice {
+  period_duration: string | null;
+  price: Price | null;
```
Comment (tonidero) [inline src/networking/responses/products-response.ts:8]: Hmm in the end, I think the `price` should never be null? And I'm not sure about the `period_duration`....

**Review by tonidero — APPROVED** [review]
Just a couple of questions but looking good!

File: src/networking/responses/products-response.ts (hunk 1 above)
Comment (nicfix) [reply src/networking/responses/products-response.ts:22]: Absolutely, changing it!

**Comment by tonidero** [comment]
I would suggest labeling this PR with `refactor` so it works with the automations, and considering this PR itself doesn't provide any new features yet.

File: src/networking/responses/products-response.ts (hunk 2 above)
Comment (nicfix) [reply src/networking/responses/products-response.ts:8]: The price is null for trials

**Comment by tonidero** [comment]
Also, might be good to change the PR title to something more user-facing since it's used for the CHANGELOG auto generation.

File: src/networking/responses/products-response.ts (hunk 2 above)
Comment (tonidero) [reply src/networking/responses/products-response.ts:8]: Ahh right makes sense. Same question about the duration though

Comment (nicfix) [reply src/networking/responses/products-response.ts:8]: In case of the period_duration that was supposed to be reused for one-offs.

Comment (tonidero) [reply src/networking/responses/products-response.ts:8]: Makes sense, thanks!

**Pull request merged**